KubernetesManifest attempts to chmod an empty string path (caused by KUBECONFIG variable)

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: KubernetesManifest

## Environment
- Server
Azure Pipelines, org `GenetecSCX`, project `SCX`, build definition `Tests.Identity.Sso`, build number `20201117.39019`

- Agent
    We use hosted, but not sure what info to give...
    ```
    Pool: Azure Pipelines
    Image: Ubuntu16
    Agent: Azure Pipelines 2
    ```

## Issue Description

The value of KUBECONFIG is an empty string, and since `"" != null => true`, we end up doing a chmod on an empty string... This issue did not happen with v 0.175.x. It only started today with 0.178.x.

The value of KUBECONFIG is set to empty by another task earlier in the job... we're investigating why this is happening, but either way I think the KubernetesManifest bake command should be resilient to this variable being an empty string.

This is very most likely related to https://github.com/microsoft/azure-pipelines-tasks/commit/ef8ef0152d9ed487d0b9b03a7646f361dd3bf215#diff-227cb7cd0f82440e51f7a323a7fd248f7a8ed02bf5c5c184b492c67c5d2ce5a2

### Task logs
```
2020-11-17T21:37:24.1728108Z ##[debug]Evaluating condition for step: 'Bake sso-tests'
2020-11-17T21:37:24.1729434Z ##[debug]Evaluating: SucceededNode()
2020-11-17T21:37:24.1730225Z ##[debug]Evaluating SucceededNode:
2020-11-17T21:37:24.1731128Z ##[debug]=> True
2020-11-17T21:37:24.1731820Z ##[debug]Result: True
2020-11-17T21:37:24.1732483Z ##[section]Starting: Bake sso-tests
2020-11-17T21:37:24.1742002Z ==============================================================================
2020-11-17T21:37:24.1742535Z Task         : Deploy to Kubernetes
2020-11-17T21:37:24.1743216Z Description  : Use Kubernetes manifest files to deploy to clusters or even bake the manifest files to be used for deployments using Helm charts
2020-11-17T21:37:24.1743853Z Version      : 0.178.0
2020-11-17T21:37:24.1744263Z Author       : Microsoft Corporation
2020-11-17T21:37:24.1744765Z Help         : https://aka.ms/azpipes-k8s-manifest-tsg
2020-11-17T21:37:24.1745372Z ==============================================================================
2020-11-17T21:37:24.1911444Z ##[debug]Using node path: /home/vsts/agents/2.177.1/externals/node/bin/node
2020-11-17T21:37:24.3083236Z ##[debug]agent.TempDirectory=/home/vsts/work/_temp
2020-11-17T21:37:24.3113130Z ##[debug]loading inputs and endpoints
2020-11-17T21:37:24.3113793Z ##[debug]loading INPUT_ACTION
2020-11-17T21:37:24.3123024Z ##[debug]loading INPUT_KUBERNETESSERVICECONNECTION
2020-11-17T21:37:24.3126801Z ##[debug]loading INPUT_NAMESPACE
2020-11-17T21:37:24.3128715Z ##[debug]loading INPUT_STRATEGY
2020-11-17T21:37:24.3130902Z ##[debug]loading INPUT_TRAFFICSPLITMETHOD
2020-11-17T21:37:24.3132356Z ##[debug]loading INPUT_PERCENTAGE
2020-11-17T21:37:24.3133992Z ##[debug]loading INPUT_BASELINEANDCANARYREPLICAS
2020-11-17T21:37:24.3135485Z ##[debug]loading INPUT_MANIFESTS
2020-11-17T21:37:24.3137254Z ##[debug]loading INPUT_RENDERTYPE
2020-11-17T21:37:24.3138900Z ##[debug]loading INPUT_DOCKERCOMPOSEFILE
2020-11-17T21:37:24.3140532Z ##[debug]loading INPUT_HELMCHART
2020-11-17T21:37:24.3142171Z ##[debug]loading INPUT_RELEASENAME
2020-11-17T21:37:24.3143734Z ##[debug]loading INPUT_OVERRIDES
2020-11-17T21:37:24.3145385Z ##[debug]loading INPUT_KUSTOMIZATIONPATH
2020-11-17T21:37:24.3146966Z ##[debug]loading INPUT_RESOURCETOPATCH
2020-11-17T21:37:24.3148497Z ##[debug]loading INPUT_RESOURCEFILETOPATCH
2020-11-17T21:37:24.3150077Z ##[debug]loading INPUT_MERGESTRATEGY
2020-11-17T21:37:24.3151687Z ##[debug]loading INPUT_SECRETTYPE
2020-11-17T21:37:24.3153448Z ##[debug]loading INPUT_ROLLOUTSTATUSTIMEOUT
2020-11-17T21:37:24.3156617Z ##[debug]loading ENDPOINT_AUTH_46c1d279-4e8d-456f-9478-e896a83aee34
2020-11-17T21:37:24.3167413Z ##[debug]loading ENDPOINT_AUTH_SCHEME_46c1d279-4e8d-456f-9478-e896a83aee34
2020-11-17T21:37:24.3168504Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_AZURETENANTID
2020-11-17T21:37:24.3171720Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_AZUREENVIRONMENT
2020-11-17T21:37:24.3179185Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_SERVICEACCOUNTNAME
2020-11-17T21:37:24.3188126Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_ROLEBINDINGNAME
2020-11-17T21:37:24.3194451Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_SECRETNAME
2020-11-17T21:37:24.3195474Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_APITOKEN
2020-11-17T21:37:24.3196468Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_46c1d279-4e8d-456f-9478-e896a83aee34_SERVICEACCOUNTCERTIFICATE
2020-11-17T21:37:24.3197265Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2020-11-17T21:37:24.3197982Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2020-11-17T21:37:24.3198716Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2020-11-17T21:37:24.3273134Z ##[debug]loaded 77
2020-11-17T21:37:24.3273742Z ##[debug]Agent.ProxyUrl=undefined
2020-11-17T21:37:24.3274335Z ##[debug]Agent.CAInfo=undefined
2020-11-17T21:37:24.3274950Z ##[debug]Agent.ClientCert=undefined
2020-11-17T21:37:24.3275586Z ##[debug]Agent.SkipCertValidation=undefined
2020-11-17T21:37:24.3446921Z ##[debug]Agent.ProxyUrl=undefined
2020-11-17T21:37:24.3447588Z ##[debug]Agent.CAInfo=undefined
2020-11-17T21:37:24.3449165Z ##[debug]Agent.ClientCert=undefined
2020-11-17T21:37:24.3451590Z ##[debug]check path : /home/vsts/work/_tasks/KubernetesManifest_dee316a2-586f-4def-be79-488a1f503dfe/0.178.0/node_modules/azure-pipelines-tool-lib/lib.json
2020-11-17T21:37:24.3452969Z ##[debug]adding resource file: /home/vsts/work/_tasks/KubernetesManifest_dee316a2-586f-4def-be79-488a1f503dfe/0.178.0/node_modules/azure-pipelines-tool-lib/lib.json
2020-11-17T21:37:24.3454233Z ##[debug]system.culture=en-US
2020-11-17T21:37:24.3859491Z ##[debug]SYSTEM_HOSTTYPE=build
2020-11-17T21:37:24.3860491Z ##[debug]System.TeamFoundationCollectionUri=https://dev.azure.com/GenetecSCX/
2020-11-17T21:37:24.3861280Z ##[debug]Build.BuildNumber=20201117.39019
2020-11-17T21:37:24.3861952Z ##[debug]Build.DefinitionName=Tests.Identity.Sso
2020-11-17T21:37:24.3862577Z ##[debug]System.DefinitionId=117
2020-11-17T21:37:24.3863236Z ##[debug]Agent.JobName=IntegrationTestsBrowser
2020-11-17T21:37:24.3863849Z ##[debug]System.TeamProject=SCX
2020-11-17T21:37:24.3864428Z ##[debug]Build.BuildId=39019
2020-11-17T21:37:24.3865024Z ##[debug]System.TeamProject=SCX
2020-11-17T21:37:24.3897287Z ##[debug]namespace=sso
2020-11-17T21:37:24.3898934Z ##[debug]containers=null
2020-11-17T21:37:24.3901852Z ##[debug]imagePullSecrets=null
2020-11-17T21:37:24.3902569Z ##[debug]manifests=/home/vsts/work/1/s
2020-11-17T21:37:24.3903169Z ##[debug]percentage=0
2020-11-17T21:37:24.3903736Z ##[debug]strategy=none
2020-11-17T21:37:24.3913680Z ##[debug]trafficSplitMethod=pod
2020-11-17T21:37:24.3914488Z ##[debug]baselineAndCanaryReplicas=1
2020-11-17T21:37:24.3915105Z ##[debug]arguments=null
2020-11-17T21:37:24.3915687Z ##[debug]secretArguments=null
2020-11-17T21:37:24.3916293Z ##[debug]secretType=dockerRegistry
2020-11-17T21:37:24.3916894Z ##[debug]secretName=null
2020-11-17T21:37:24.3917494Z ##[debug]dockerRegistryEndpoint=null
2020-11-17T21:37:24.3918090Z ##[debug]rolloutStatusTimeout=0
2020-11-17T21:37:24.3946298Z ##[debug]System.TeamFoundationCollectionUri=https://dev.azure.com/GenetecSCX/
2020-11-17T21:37:24.3947060Z ##[debug]System.HostType=build
2020-11-17T21:37:24.3947734Z ##[debug]System.DefaultWorkingDirectory=/home/vsts/work/1/s
2020-11-17T21:37:24.3949092Z ##[debug]Build.SourceBranchName=ls-to-the-rescue
2020-11-17T21:37:24.3949709Z ##[debug]Build.Repository.Provider=TfsGit
2020-11-17T21:37:24.3950718Z ##[debug]Build.Repository.Uri=https://GenetecSCX@dev.azure.com/GenetecSCX/SCX/_git/Tests.Identity.Sso
2020-11-17T21:37:24.4024801Z ##[debug]agent.proxyurl=undefined
2020-11-17T21:37:24.4025531Z ##[debug]VSTS_ARM_REST_IGNORE_SSL_ERRORS=undefined
2020-11-17T21:37:24.4026705Z ##[debug]AZURE_HTTP_USER_AGENT=VSTS_799f22d4-b8e8-49cf-bb91-9e0d6bb539a1_build_117_0
2020-11-17T21:37:24.4123330Z ##[debug]check path : /home/vsts/work/_tasks/KubernetesManifest_dee316a2-586f-4def-be79-488a1f503dfe/0.178.0/task.json
2020-11-17T21:37:24.4124622Z ##[debug]adding resource file: /home/vsts/work/_tasks/KubernetesManifest_dee316a2-586f-4def-be79-488a1f503dfe/0.178.0/task.json
2020-11-17T21:37:24.4125592Z ##[debug]system.culture=en-US
2020-11-17T21:37:24.4143358Z ##[debug]check path : /home/vsts/work/_tasks/KubernetesManifest_dee316a2-586f-4def-be79-488a1f503dfe/0.178.0/node_modules/azure-pipelines-tasks-kubernetes-common-v2/module.json
2020-11-17T21:37:24.4146007Z ##[debug]adding resource file: /home/vsts/work/_tasks/KubernetesManifest_dee316a2-586f-4def-be79-488a1f503dfe/0.178.0/node_modules/azure-pipelines-tasks-kubernetes-common-v2/module.json
2020-11-17T21:37:24.4147152Z ##[debug]system.culture=en-US
2020-11-17T21:37:24.4147778Z ##[debug]action=bake
2020-11-17T21:37:24.4155523Z ##[debug]renderType=helm
2020-11-17T21:37:24.4160741Z ##[debug]KUBECONFIG=
2020-11-17T21:37:24.4199651Z ##[debug]task result: Failed
2020-11-17T21:37:24.4246468Z ##[error]ENOENT: no such file or directory, chmod ''
2020-11-17T21:37:24.4255666Z ##[debug]Processed: ##vso[task.issue type=error;]ENOENT: no such file or directory, chmod ''
2020-11-17T21:37:24.4257651Z ##[debug]Processed: ##vso[task.complete result=Failed;]ENOENT: no such file or directory, chmod ''
2020-11-17T21:37:24.4259972Z ##[section]Finishing: Bake sso-tests
```

--- a/Tasks/KubernetesManifestV0/src/actions/bake.ts
+++ b/Tasks/KubernetesManifestV0/src/actions/bake.ts
@@ -38,7 +38,7 @@ class HelmRenderEngine extends RenderEngine {
     public bake = async (): Promise<any> => {
         // Helm latest releases require restricted permissions on Kubeconfig
         const kubeconfigPath = tl.getVariable('KUBECONFIG');
-        if (kubeconfigPath != null)
+        if (kubeconfigPath)
             fs.chmodSync(kubeconfigPath, '600');
         const helmPath = await helmutility.getHelm();
         const helmCommand = new Helm(helmPath, TaskParameters.namespace);

--- a/Tasks/KubernetesManifestV0/task.json
+++ b/Tasks/KubernetesManifestV0/task.json
@@ -13,7 +13,7 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 0,
-        "Minor": 178,
+        "Minor": 179,
         "Patch": 0
     },
     "demands": [],

--- a/Tasks/KubernetesManifestV0/task.loc.json
+++ b/Tasks/KubernetesManifestV0/task.loc.json
@@ -13,7 +13,7 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 0,
-    "Minor": 178,
+    "Minor": 179,
     "Patch": 0
   },
   "demands": [],

